app/services/valid_test_data_generators/applications_populater.rb

```diff
@@ -0,0 +1,103 @@
+
+module ValidTestDataGenerators
+  class ApplicationsPopulater
+    class << self
+      def populate(lead_provider:, cohort:, number_of_participants: 50)
+        new(lead_provider:, cohort:, number_of_participants:).populate
+      end
+    end
+
+    def populate
+      return unless Rails.env.in?(%w[development separation])
+
+      logger.info "ApplicationsPopulater: Started!"
+
+      ActiveRecord::Base.transaction do
+        create_participants!
+      end
+
+      logger.info "ApplicationsPopulater: Finished!"
+    end
+
+  private
+
+    attr_reader :lead_provider, :cohort, :number_of_participants, :courses, :logger
+
+    def initialize(lead_provider:, cohort:, number_of_participants:, logger: Rails.logger)
+      @lead_provider = lead_provider
+      @cohort = cohort
+      @number_of_participants = number_of_participants
+      @logger = logger
+      @courses = Course.all.reject { |c| c.identifier == "npq-early-headship-coaching-offer" }
+    end
+
+    def create_participants!
+      number_of_participants.times { create_participant(school: School.open.order("RANDOM()").first) }
+    end
+
+    def create_participant(school:)
+      trn = TRNGenerator.next
+      course = courses.sample
+
+      user = FactoryBot.create(:user,
+                               :with_get_an_identity_id,
+                               :with_random_name,
+                               date_of_birth: Date.new(1990, 1, 1),
+                               trn:,
+                               trn_verified: true,
+                               trn_auto_verified: true)
+
+      application = FactoryBot.create(:application,
+                                      :pending,
+                                      :with_random_work_setting,
+                                      school:,
+                                      lead_provider:,
+                                      user:,
+                                      cohort:,
+                                      course:,
+                                      headteacher_status: Application.headteacher_statuses.keys.sample,
+                                      lead_mentor: [true, false].sample,
+                                      eligible_for_funding: [true, false].sample,
+                                      itt_provider: IttProvider.currently_approved.order("RANDOM()").first)
+
+      methods = %i[accept_application reject_application]
+
+      return if [true, false].sample
+
+      send(methods.sample, application)
+
+      return if [true, false].sample
+
+      application = FactoryBot.create(:application,
+                                      :pending,
+                                      :with_random_work_setting,
+                                      school:,
+                                      lead_provider:,
+                                      user:,
+                                      cohort:,
+                                      course:,
```

Comment:
could we use a different course here, maybe a random course except for the one selected previously to avoid having multiple applications with the same course accepted?